ci/cloudfront-orchestration/waf/template.yaml

```diff
@@ -0,0 +1,212 @@
+            Block: {}
+          Priority: 5
+          Statement:
+            AndStatement:
+              Statements:
+                - SizeConstraintStatement:
+                    ComparisonOperator: "GT"
+                    FieldToMatch:
+                      QueryString: {}
+                    Size: 2048
+                    TextTransformations:
+                      - Priority: 0
+                        Type: "NONE"
+                - NotStatement:
+                    Statement:
+                      ByteMatchStatement:
+                        FieldToMatch:
+                          UriPath: {}
+                        PositionalConstraint: "EXACTLY"
+                        SearchString: "/authorize"
+                        TextTransformations:
+                          - Priority: 0
+                            Type: "LOWERCASE"
+          VisibilityConfig:
+            CloudWatchMetricsEnabled: true
+            MetricName: !Sub "${Environment}OidcWafQueryParamSet"
+            SampledRequestsEnabled: true
+        - Name: !Sub "${Environment}-extended-query-param-limit"
+          Action:
+            Block: {}
+          Priority: 6
+          Statement:
+            SizeConstraintStatement:
+              ComparisonOperator: "GT"
+              FieldToMatch:
+                QueryString: {}
+              Size: 4096
+              TextTransformations:
+                - Priority: 0
+                  Type: "NONE"
+          VisibilityConfig:
+            CloudWatchMetricsEnabled: true
+            MetricName: !Sub "${Environment}OidcWafAuthorizeQueryParamSet"
+            SampledRequestsEnabled: true
+      VisibilityConfig:
+        CloudWatchMetricsEnabled: true
+        MetricName: !Sub "${Environment}OidcWafRules"
+        SampledRequestsEnabled: true
+
+  LogKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: Encrypts logs for the OIDC WAF attached to Cloudfront
+      KeyPolicy:
+        Version: 2012-10-17
+        Id: OidcWafLogKeyPolicy
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action: "kms:*"
+            Resource: "*"
+          - Sid: Permit Cloudwatch
+            Effect: Allow
+            Principal:
+              Service: !Sub "logs.${AWS::Region}.amazonaws.com"
+            Action:
+              - "kms:Encrypt*"
+              - "kms:Decrypt*"
+              - "kms:ReEncrypt*"
+              - "kms:GenerateDataKey*"
+              - "kms:Describe*"
+            Resource: "*"
+            Condition:
+              ArnEquals:
+                "kms:EncryptionContext:aws:logs:arn": !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:aws-waf-logs-cf-oidc-${Environment}"
+
+
+  LogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "aws-waf-logs-cf-oidc-${Environment}"
+      RetentionInDays: 30
```

Comment:
Is there a reason why `${Environment}` is at the end of the name here but for other names it is at the beginning?

Reply:
WAF log groups must start with `aws-waf-logs` because of some silly amazon rules

Reply:
https://github.com/govuk-one-login/authentication-api/pull/4321#discussion_r1576018223